src/bsp/src/stm32/include/Decawave.h

```diff
@@ -0,0 +1,163 @@
+
+    /**
+     * @brief Starts listening for a packet at a given timestamp without blocking. The user should
+     * poll the frame.status to see if transmission has been received.
+     * @param frame[out] Variable in which all frame data will be put once packet is received
+     * @param timeoutUs Timeout in microseconds. 0 to block indefinitely
+     * @param rxStartTime DW timestamp at which to start the RX
+     */
+    void receiveAsyncDelayed(UWBRxFrame& frame, uint16_t timeoutUs, uint64_t rxStartTime);
+
+    /**
+     * @brief Transmits data over UWB
+     * @param buf Buffer to transmit
+     * @param length Length of the data
+     * @return True if successful, false otherwise.
+     */
+    bool transmit(uint8_t* buf, uint16_t length);
+
+    /**
+     * @brief Transmits data over UWB at a given timestamp
+     * @param buf Buffer to transmit
+     * @param length Length of the data
+     * @param txTimestamp DW timestamp at which to start the transmission
+     * @return True if successful, false otherwise.
+     */
+    bool transmitDelayed(uint8_t* buf, uint16_t length, uint64_t txTimestamp);
+
+    /**
+     * @brief Transmits data over UWB and blocks until response is received
+     * @param buf Buffer to transmit
+     * @param length Length of the data
+     * @param rxAfterTxTimeUs Number of microseconds between the TX and the start of the RX
+     * @param frame[out] Variable in which all frame data will be put once packet is received
+     * @param rxTimeoutUs Timeout in microseconds. 0 to block indefinitely
+     * @return True if successful, false otherwise.
+     */
+    bool transmitAndReceive(uint8_t* buf,
+                            uint16_t length,
+                            uint32_t rxAfterTxTimeUs,
+                            UWBRxFrame& frame,
+                            uint16_t rxTimeoutUs);
+
+    /**
+     * @brief Transmits data over UWB and blocks until response is received
+     * @param buf Buffer to transmit
+     * @param length Length of the data
+     * @param txTimestamp DW timestamp at which to start the transmission
+     * @param rxAfterTxTimeUs Number of microseconds between the TX and the start of the RX
+     * @param frame[out] Variable in which all frame data will be put once packet is received
+     * @param rxTimeoutUs Timeout in microseconds. 0 to block indefinitely
+     * @return True if successful, false otherwise.
+     */
+    bool transmitAndReceiveDelayed(uint8_t* buf,
+                                   uint16_t length,
+                                   uint64_t txTimestamp,
+                                   uint32_t rxAfterTxTimeUs,
+                                   UWBRxFrame& frame,
+                                   uint16_t rxTimeoutUs);
+
+    friend void rxCallback(const dwt_cb_data_t* callbackData, void* context);
+    friend void isrCallback(void* context);
+    friend void rxAsyncTask(void* context);
```

Comment:
You could declare them static instead of friends, this would allow you to access private members I think even when you cast the context. And you could  probably set them private too